framework/src/executors/Executor.C

```diff
@@ -32,12 +34,12 @@ Executor::Executor(const InputParameters & parameters)
 {
   if (!parameters.isParamSetByUser("begin_exec_flag"))
   {
-    _begin_flag = ExecFlagType("exec_" + _name + "_begin");
+    _begin_flag = moose::internal::getExecFlagRegistry().registerFlag("exec_" + _name + "_begin");
```

Comment:
```suggestion
    _begin_flag = defineExecFlag("exec_" + _name + "_begin");
```

Am I using that right?

Reply:
Yep. I like this - expose fewer internals when possible